worlds/factorio/Technologies.py

```diff
@@ -203,18 +207,18 @@ def __init__(self, name, categories):
 # add uranium mining to logic graph. TODO: add to automatic extractor for mod support
 raw_recipes = recipes_future.result()
 del recipes_future
-
-raw_recipes["uranium-ore"] = {
-    "ingredients": {"sulfuric-acid": 1},
-    "products": {"uranium-ore": 1},
-    "category": "mining",
-    "energy": 2
-}
-
-# raw_recipes["iron-ore"] = {"ingredients": {}, "products": {"iron-ore": 1}, "category": "mining", "energy": 2}
-# raw_recipes["copper-ore"] = {"ingredients": {}, "products": {"copper-ore": 1}, "category": "mining", "energy": 2}
-# raw_recipes["coal-ore"] = {"ingredients": {}, "products": {"coal": 1}, "category": "mining", "energy": 2}
-# raw_recipes["stone"] = {"ingredients": {}, "products": {"coal": 1}, "category": "mining", "energy": 2}
+for resource_name, resource_data in resources_future.result().items():
+    raw_recipes[f"mining-{resource_name}"] = current_recipe = {
+        "ingredients": {},
+        "products": {},
+        "energy": resource_data["mining_time"],
+        "category": resource_data["category"]
+    }
+    for name, data in resource_data["products"].items():
+        current_recipe["products"].update({data["name"]: data["amount"]})
+    if "required_fluid" in resource_data:
+        current_recipe["ingredients"] = {resource_data["required_fluid"]: resource_data["fluid_amount"]}
```

Comment:
```suggestion
    raw_recipes[f"mining-{resource_name}"] = current_recipe = {
        "ingredients": {resource_data["required_fluid"]: resource_data["fluid_amount"]}
        if "required_fluid" in resource_data else {},
        "products": {data["name"]: data["amount"] for data in resource_data["products"].values()},
        "energy": resource_data["mining_time"],
        "category": resource_data["category"]
    }
```

Reply:
sure, though then we don't need to bind to current_recipe either anymore.

Reply:
oh, yeah, missed that. changes look good otherwise, i think